### ISSUE TITLE
System.NotSupportedException after upgrading to 5.2.0

### ISSUE THREAD

**Issue by celluj34** [dotnet/SqlClient#2407]
### Describe the bug

We recently upgraded from SqlClient 5.1.5 to 5.2.0. We are disabling invariant mode by setting `<InvariantGlobalization>false</InvariantGlobalization>` in our Directory.Build.props file. However we are seeing exceptions on startup due to `Unhandled exception. System.NotSupportedException: Globalization Invariant Mode is not supported`.

I have verified we are including the `tzdata` and `icu-libs` in our container image. Interestingly, we do not see the error when running locally.

We are currently running in production with 5.1.5, but after upgrade to 5.2.0, we are seeing startup errors.

### Expected behavior
I expect invariant mode to be determined correctly.

### Further technical details
Microsoft.Data.SqlClient version: (found on the nuget or Microsoft.Data.SqlClient.dll)
.NET target: .Net 7 and .Net 8
SQL Server version: Azure hosted SQL (not sure the exact version)
Operating system: Linux containers `mcr.microsoft.com/dotnet/runtime-deps:8.0.0-alpine3.18-amd64` and `mcr.microsoft.com/dotnet/runtime-deps:7.0.16-alpine3.19-amd64`

**Additional context**

**Comment by DavoudEshtehari** [comment]
Could you provide us with a simple repro application along with the steps to reproduce the issue?

**Comment by celluj34** [comment]
Yes, I will see if I can have that in the next day or two.

**Comment by celluj34** [comment]
FWIW if I add the `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=false` env variable it does start up without issue.

**Comment by celluj34** [comment]
I've included the `icu-data-full` package as well as the previous env variable. I believe these two together are good enough. This is good enough for me for now, even if it does bloat our images somewhat. (170 => 200MB)

OLD

```dockerfile
<etc>

RUN apk add --no-cache \
  tzdata \
  icu-libs

ENV DOTNET_RUNNING_IN_CONTAINER=true \
  TZ=America/Chicago \
  ASPNETCORE_URLS=http://+:8080

<etc>
```

NEW
```dockerfile
<etc>

RUN apk add --no-cache \
  tzdata \
  icu-data-full \
  icu-libs

ENV DOTNET_RUNNING_IN_CONTAINER=true \
  DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=false \
  LC_ALL=en_US.UTF-8 \
  LANG=en_US.UTF-8 \
  TZ=America/Chicago \
  ASPNETCORE_URLS=http://+:8080

<etc>
```

**Comment by iem365-michael** [comment]
Not sure this should be closed, we have seen exactly the same behaviour and cannot believe it is expected.

Upgrading from from SqlClient 5.1.5 to 5.2.0 results in the error "Globalization Invariant Mode is not supported." in containers that were working previously.

As above adding "ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=false" to our docker files has resolved the issue.

Before 5.2.0 using <InvariantGlobalization>false</InvariantGlobalization> in the csproj was sufficient.

**Comment by celluj34** [comment]
@iem365-michael I closed it because tbh I didn't have the time to minimize the code for an MRE, and I found a workaround that was good enough for me after reading other issues and microsoft docs. They changed how they were detecting global invariant mode, so additionally including the env variable is required(?) now.

I can re-open but If you'd be able to help repro with a minimum repro that would be a big help.
